blockmanager.go

```diff
@@ -931,38 +937,36 @@ func (b *blockManager) getCheckpointedCFHeaders(checkpoints []*chainhash.Hash,
 	// all at once. This message will distributed the header requests
 	// amongst all active peers, effectively sharding each query
 	// dynamically.
-	b.cfg.queryBatch(
-		queryMsgs,
+	headerChan := make(chan *wire.MsgCFHeaders, len(queryMsgs))
 
-		// Callback to process potential replies. Always called from
-		// the same goroutine as the outer function, so we don't have
-		// to worry about synchronization.
-		func(sp *ServerPeer, query wire.Message,
-			resp wire.Message) bool {
+	var msgs []*query.Query
+	for _, m := range queryMsgs {
+		handleResp := func(query, resp wire.Message,
```

Comment:
In order to chop this method up a bit, we can instead make this function closure an actual function, and parameterize it with the request context.

Reply:
Extracted: https://github.com/lightninglabs/neutrino/pull/179/commits/04aa607a3914fda3b1c3ed9ff7dd603a410167cb